### PR TITLE
reports batch fill metrics again

### LINKED ISSUE #71: New reader/batcher doesn't update batch.fill metrics
This was missing when I rolled the batcher into the reader. :-(

### PR DESCRIPTION
**Pull request by voidlock**
Fixes #71 by tracking the time it takes to fill batches again.

### CODE REVIEW

File: reader.go [hunk 1]
```diff
@@ -90,9 +91,11 @@ func (rdr *LogLineReader) ReadLines() error {
 			rdr.linesRead.Inc(1)
 			rdr.mu.Lock()
 			if full := rdr.b.Add(LogLine{line, currentLogTime}); full {
+				rdr.batchFillTime.UpdateSince(now)
```
Comment (rsampaio) [inline reader.go]: Could we use `time.Now()` directly as the argument of .UpdateSince without a variable ?

File: reader.go [hunk 2]
```diff
@@ -90,9 +91,11 @@ func (rdr *LogLineReader) ReadLines() error {
...
 				rdr.deliverOrDropCurrent()
 			}
 			if rdr.b.MsgCount() == 1 { // First line so restart the timer
+				now = time.Now()
```
Comment (rsampaio) [inline reader.go]: If the answer for my last comment is yes we could remove this assignment from here.

**Comment by rsampaio** [comment]
@voidlock I created a deb with the diff below and I can confirm it is enough to bring the batch.fill metrics back:

```diff
diff --git a/reader.go b/reader.go
index b61a814..e83d495 100644
--- a/reader.go
+++ b/reader.go
@@ -90,6 +90,7 @@ func (rdr *LogLineReader) ReadLines() error {
                        rdr.linesRead.Inc(1)
                        rdr.mu.Lock()
                        if full := rdr.b.Add(LogLine{line, currentLogTime}); full {
+                               rdr.batchFillTime.UpdateSince(time.Now())
                                rdr.deliverOrDropCurrent()
                        }
                        if rdr.b.MsgCount() == 1 { // First line so restart the timer

```

This deb is now installed in our staging environemtn FWIW.

**Comment by voidlock** [comment]
@rsampaio I don't think that diff is what we need. UpdateSince [neatly wraps](https://github.com/heroku/log-shuttle/blob/378e42873ec596313441cff5c1baf8212d9cc6f0/vendor/github.com/rcrowley/go-metrics/timer.go#L228) `time.Since(timestamp)` in some mutexes. So by passing `time.Now()` it will never calculate the time it has taken to fill the batch.

**Comment by rsampaio** [comment]
@voidlock I still think the effect is the same, UpdateSince expects an instance of time.Time and that is what time.Now() function returns: https://godoc.org/time#Now

**Comment by rsampaio** [comment]
:+1: 

disregard my comments, my mistake.

**Comment by freeformz** [comment]
LGTM. Merging

**Pull request merged**